### PR TITLE
Move to ginkgo v2

### PR DESCRIPTION
**Pull request by nunnatsa**
Moving to ginkgo v2 allows filtering by labels. We want this feature for the e2e tests.

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release notes**:

```release-note
None
```

### CODE REVIEW

File: controllers/kubevirtmachine_controller_test.go [hunk 1]
```diff
@@ -298,12 +304,11 @@ var _ = Describe("reconcile a kubevirt machine", func() {
...
-		machineMock.EXPECT().Delete().Return(nil).Times(2)
 		machineMock.EXPECT().Exists().Return(true).Times(1)
-		machineMock.EXPECT().IsReady().Return(false).Times(2)
-		machineMock.EXPECT().Address().Return("1.1.1.1").Times(1)
-		machineMock.EXPECT().SupportsCheckingIsBootstrapped().Return(false).Times(1)
-		machineMock.EXPECT().GenerateProviderID().Return("abc", nil).Times(1)
+		machineMock.EXPECT().IsReady().Return(false).AnyTimes()
+		machineMock.EXPECT().Address().Return("1.1.1.1").AnyTimes()
+		machineMock.EXPECT().SupportsCheckingIsBootstrapped().Return(false).AnyTimes()
+		machineMock.EXPECT().GenerateProviderID().Return("abc", nil).AnyTimes()
```
Comment (davidvossel) [inline controllers/kubevirtmachine_controller_test.go:311]: does `Times()` not work anymore with v2? If so, that's not great

File: controllers/kubevirtmachine_controller_test.go [hunk 2]
```diff
@@ -569,7 +574,6 @@ var _ = Describe("reconcile a kubevirt machine", func() {
 
 			setupClient(kubevirt.DefaultMachineFactory{}, objects)
 
-			infraClusterMock.EXPECT().GenerateInfraClusterClient(kubevirtMachine.Spec.InfraClusterSecretRef, kubevirtMachine.Namespace, machineContext.Context).Return(fakeClient, kubevirtMachine.Namespace, nil)
```
Comment (davidvossel) [inline controllers/kubevirtmachine_controller_test.go:572]: this change and similar changes makes it look like there are conditions that we no longer test using `v2`

**Review by davidvossel — COMMENTED** [review]
there are some changes in here that look like they alter how the unit tests validate functionality. If v2 can't maintain existing compatibility with what we have, then we can't do a swap and replace.

File: controllers/kubevirtmachine_controller_test.go (hunk 1 above)
Comment (nunnatsa) [reply controllers/kubevirtmachine_controller_test.go:311]: They do. They works too good. It now enfoced by ginkgo v2. I could changed it to any value I wanted in v1, and the test passed.

File: controllers/kubevirtmachine_controller_test.go (hunk 2 above)
Comment (nunnatsa) [reply controllers/kubevirtmachine_controller_test.go:572]:
```
missing call(s) to *mock.MockInfraCluster.GenerateInfraClusterClient(is equal to nil (*v1.ObjectReference), is equal to  (string), is equal to context.Background (*context.emptyCtx))
```

File: controllers/kubevirtmachine_controller_test.go (hunk 1 above)
Comment (davidvossel) [reply controllers/kubevirtmachine_controller_test.go:311]: lol... you're right. I can change the Times() to anything and it passes for me locally to with v1. ha. that's useless.

File: controllers/kubevirtmachine_controller_test.go (hunk 2 above)
Comment (davidvossel) [reply controllers/kubevirtmachine_controller_test.go:572]: yup, wow, i removed this line locally using v1 and the tests still passed.

**Review by davidvossel — APPROVED** [review]
/lgtm
/approve

thanks!

**Pull request merged**